cg/clients/freshdesk/freshdesk_client.py

```diff
@@ -11,28 +18,51 @@
 class FreshdeskClient:
     """Client for communicating with the freshdesk REST API."""
 
-    def __init__(self):
-        self.headers = None
-        self.api_key = None
-        self.url = None
-
-    def init_app(self, url: str, api_key: str):
-        """Set up the client."""
-        self.url = url
+    def __init__(self, base_url: str, api_key: str):
+        self.base_url = base_url
         self.api_key = api_key
-        self.headers = {"Content-Type": "application/json"}
+        self.session = self._configure_session()
+
+    def __enter__(self) -> "FreshdeskClient":
+        return self
 
-    @property
-    def auth_header(self):
-        return self.api_key, "X"
+    def __exit__(self, exc_type, exc_value, traceback) -> None:
+        self.session.close()
 
     def create_ticket(self, ticket: TicketCreate) -> TicketResponse:
         """Create a ticket."""
         LOG.debug(ticket.model_dump_json())
-        response = requests.post(
-            url=f"{self.url}{EndPoints.TICKETS}",
-            headers=self.headers,
-            auth=self.auth_header,
-            json=ticket.model_dump(exclude_none=True),
+        try:
+            response: Response = self.session.post(
+                url=f"{self.base_url}{EndPoints.TICKETS}",
+                json=ticket.model_dump(exclude_none=True),
+            )
+            response.raise_for_status()
+            ticket_response: TicketResponse = TicketResponse.model_validate(response.json())
+        except RequestException as error:
```

Comment:
What errors will be caught here? raise_for_status raises an HTTPException.

Reply:
`RequestException` is a parent to `HTTPErrror` so it will catch more stuff as well. Such as `ConnectionError`, `Timeout` and `ProxyError`. I think it makes sense to catch it quite wide